llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5897,6 +5898,7 @@ bool SPIRVToLLVM::translate(ExecutionModel entryExecModel, const char *entryName
   if (!m_entryTarget)
     return false;
 
+  m_shaderStage = convertToStageShage(entryExecModel);
```

Comment:
Could we instead store entryExecModel in a new member field? Then future commits that use it would compare against the SPIR-V execution model, instead of the LLPC shader stage.